doc/notes/11455_deployments/11455_export_depl.dnt.adoc

```diff
@@ -0,0 +1,227 @@
+= Export Deployment (Build Set)
+
+xtUML Project Design Note
+
+== 1 Abstract
+
+This note describes the design to properly export Deployments, with sequenced terminators, to WASL.
+
+== 2 Introduction and Background
+
+See <<dr-2>>.
+
+In addition sequenced terminator support was added in Bridgepoint editor according to <<dr-3>>.
+
+== 3 Requirements
+
+See <<dr-2>>.
+
+== 4 Analysis
+
+See <<dr-2>>.
+
+
+== 5 Design
+
+=== 5.1 Add metamodel to mcooa
+
+The D_TSS and D_SIS classes and associations from <<dr-3>> are added to the mcooa model.
+
+=== 5.2 WASL Output
+
+==== 5.2.1 Add key letter support to masl
+
+WASL output makes use of project key letters.  <<dr-3>> added key letters to Deployments to support this.  In order to compile the model we must also add key letter support to the masl model project element.  The maslout model is modified to pass in key letters from xtUML through the populate port.  project.populate() is modified to accept a keyletters param and set it on the project instance created.
+
+==== 5.2.2 WASL descriptive files
+
+WASL build sets have a descriptive file named after the project key letters.  The file has an extension of dmp.
+
+This description contains build set information comprised of in place information as well as pointers to other files.  The other files contain further build set information, and in some cases the actual bridge implementations.
+
+==== 5.2.2.3 Build set/project information
+
+```
+$BUILD_SET_NAME <project.name>
+$BUILD_SET_VERSION 0
+$PROJECT_KEY_LETTER <project.keyletters>
+$PROJECT_VERSION 0
+```
+
+Added templates:
+
+* t.project_begin.wasl
+* t.project_end.wasl
+
+A new operation is added to project, render_wasl.  The genfile.render_wasl operation is updated to move all project related logic to the new operation.
+
+Calls to the two templates are added to render_wasl.  The project_end is output after all required description data is output.  For all cases below that are output to the description file, they are output inbetween the start and end project templates.
+
+==== 5.2.2.4 Build set tags
+
+Tags will not be used, so this will always be empty.
+
+Added templates:
+
+* t.build_set_tags_begin.wasl
+* t.build_set_tags_end.wasl
+
+```
+$START_BUILD_SET_TAGS
+$END_BUILD_SET_TAGS
+```
+
+==== 5.2.2.5 Scenario list
+
+A pointer to a file containing all scenarios in the build set.
+
+Added template:
+
+* t.scenario_list_spec.wasl
+
+```
+$START_SCENARIO_LIST
+Events.scn
+$END_SCENARIO_LIST
+```
+
+==== 5.2.2.5.1 <project.keyletters>.scn file
+
+This file is output with an ordered list of sequenced terminators.  The list of these terminators is collected in the changes described in <<scenario_file>>.
```

Comment:
When you say 'sequences terminators', I think you mean 'sequenced terminator services'.
terminator maps to port
terminator service maps to message (operation), which is also called a bridge.

Reply:
And which here is also called a scenario.  The name overloading is pretty going to keep tripping us up.  :)

Reply:
Updated as it should be sequenced terminator service.